test/stdlib/Integer_test.rb

```diff
@@ -1,398 +1,811 @@
 require_relative "test_helper"
 
-class IntegerTest < StdlibTest
-  target Integer
+class IntegerSingletonTest < Test::Unit::TestCase
+  include TestHelper
+
+  testing "singleton(::Integer)"
 
   def test_sqrt
-    Integer.sqrt(4)
-    Integer.sqrt(4.0)
-    Integer.sqrt(4/1r)
-    Integer.sqrt(ToInt.new)
+    with_int do |int|
+      assert_send_type  '(int) -> Integer',
+                        Integer, :sqrt, int
+    end
   end
 
-  def test_modulo
-    3 % 1
-    3 % 1.1
-    3 % 1.5r
-  end
-
-  def test_bitwise_ops
-    3 & 1
-    1 ^ 3
-  end
-
-  def test_calc
-    3 * 1
-    3 * 1.0
-    3 * (1r/3)
-
-    3 - 1
-    3 - 0.1
-    3 - 1r
-    3 - 1.to_c
-
-    1 ** 1
-    2 ** 2.1
-    3 ** 3r
-    3 ** 10.to_c
-
-    3 + 1
-    3 + 1.0
-    3 + (1r/3)
-    3 + 10.to_c
-
-    3 / 1
-    3 / 1.0
-    3 / (1r/3)
-    30 / 10.to_c
-  end
-
-  def test_compare
-    3 < 1
-    3 < 1.0
-    3 < 1r
-
-    1 > 1
-    1 > 1.0
-    1 > 1r
-
-    1 <= 3
-    1 <= 1.3
-    1 <= 3r
-
-    1 >= 3
-    1 >= 1.3
-    1 >= 3r
-
-    1 <=> 1
-    1 <=> 1.0
-    1 <=> 3r
-
-    3 === 3.0
-    3 === ""
-  end
-
-  def test_shift
-    1 << 30
-    1 << 30.to_f
-    1 << ToInt.new
-
-    1 >> 30
-    1 >> 30.to_f
-    1 >> ToInt.new
-  end
-
-  def test_aref
-    3[0]
-    3[0.3]
-    3[ToInt.new]
-
-    3[1,2]
-    3[1...3]
-  end
-
-  def test_to_s
-    1.to_s
-    1.to_s(2)
-    1.to_s(3)
-    1.to_s(4)
-    1.to_s(5)
-    1.to_s(6)
-    1.to_s(7)
-    1.to_s(8)
-    1.to_s(9)
-    1.to_s(10)
-    1.to_s(11)
-    1.to_s(12)
-    1.to_s(13)
-    1.to_s(14)
-    1.to_s(15)
-    1.to_s(16)
-    1.to_s(17)
-    1.to_s(18)
-    1.to_s(19)
-    1.to_s(20)
-    1.to_s(21)
-    1.to_s(22)
-    1.to_s(23)
-    1.to_s(24)
-    1.to_s(25)
-    1.to_s(26)
-    1.to_s(27)
-    1.to_s(28)
-    1.to_s(29)
-    1.to_s(30)
-    1.to_s(31)
-    1.to_s(32)
-    1.to_s(33)
-    1.to_s(34)
-    1.to_s(35)
-    1.to_s(36)
-    30.to_s(ToInt.new)
-  end
-
-  def test_abs_abs2
-    3.abs
-    3.abs2
-  end
+  def test_try_convert
+    with_int do |int|
+      assert_send_type  '(int) -> Integer',
+                        Integer, :try_convert, int
+    end
 
-  def test_allbits?
-    1.allbits?(1)
-    2.allbits?(1)
-    3.allbits?(ToInt.new)
+    with_untyped do |untyped|
+      assert_send_type  '(untyped) -> Integer?',
+                        Integer, :try_convert, untyped
+    end
   end
+end
+
+class IntegerInstanceTest < Test::Unit::TestCase
+  include TestHelper
 
-  def test_angle
-    3.angle()
+  testing "::Integer"
+
+  def with_random_Integers
```

Comment:
Instead of using hardcoded integers, I figured generally using a range of integers would cover more cases.